### PR TITLE
implementation models and dao

### PR DESCRIPTION
**Pull request by Daniel1004K**

### CODE REVIEW

File: src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java [hunk 1]
```diff
@@ -12,21 +15,64 @@ public CommentDaoImpl(SessionFactory sessionFactory) {
...
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.delete(entity);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't delete entity " + entity, e);
```
Comment (PashaTerletskyi) [inline src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java:71]:
```suggestion
            throw new RuntimeException("Can't delete comment " + entity, e);
```

File: src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java [hunk 2]
```diff
@@ -12,21 +15,63 @@ public MessageDaoImpl(SessionFactory sessionFactory) {
...
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.delete(entity);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't delete entity " + entity, e);
```
Comment (PashaTerletskyi) [inline src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java:70]:
```suggestion
            throw new RuntimeException("Can't delete message " + entity, e);
```

File: src/main/java/core/basesyntax/HibernateUtil.java [hunk 3]
```diff
@@ -13,7 +13,7 @@ private static SessionFactory initSessionFactory() {
         try {
             return new Configuration().configure().buildSessionFactory();
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            throw new RuntimeException("Configuration isn't completed.",e);
```
Comment (PashaTerletskyi) [inline src/main/java/core/basesyntax/HibernateUtil.java:16]:
```suggestion
            throw new RuntimeException("Configuration isn't completed.", e);
```

File: src/main/java/core/basesyntax/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -12,21 +15,63 @@ public UserDaoImpl(SessionFactory sessionFactory) {
...
+        }
     }
 
     @Override
     public User get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.get(User.class, id);
+        } catch (Exception e) {
+            throw new RuntimeException("User with id = " + id + "doesn't exist", e);
```
Comment (PashaTerletskyi) [inline src/main/java/core/basesyntax/dao/impl/UserDaoImpl.java:43]:
```suggestion
            throw new RuntimeException("User with id = " + id + " doesn't exist", e);
```

**Review by PashaTerletskyi — APPROVED** [review]

File: src/main/resources/hibernate.cfg.xml [hunk 5]
```diff
@@ -1,18 +1,17 @@
...
-        <property name="connection.url">jdbc:mysql://localhost/cinema?serverTimezone=UTC</property>
-        <property name="connection.driver_class">com.mysql.cj.jdbc.Driver</property>
-        <property name="connection.username">ma_admin</property>
-        <property name="connection.password">1234</property>
-        <property name="show_sql">true</property>
-        <property name="hbm2ddl.auto">update</property>
-
+  <session-factory>
+    <property name="hibernate.dialect">org.hibernate.dialect.MySQL5Dialect</property>
+    <property name="connection.url">jdbc:mysql://localhost8080/cinema?serverTimezone=UTC</property>
```
Comment (KseniiaMakarova) [inline src/main/resources/hibernate.cfg.xml:4]: will this url work?

File: src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java [hunk 6]
```diff
@@ -12,21 +15,64 @@ public CommentDaoImpl(SessionFactory sessionFactory) {
...
+        }
     }
 
     @Override
     public Comment get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.get(Comment.class, id);
+        } catch (Exception e) {
+            throw new RuntimeException("Comment with id = " + id + "isn't exist", e);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java:43]:
```suggestion
            throw new RuntimeException("Comment with id = " + id + " doesn't exist", e);
```
same everywhere

File: src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java [hunk 7]
```diff
@@ -12,21 +15,64 @@ public CommentDaoImpl(SessionFactory sessionFactory) {
...
+        }
     }
 
     @Override
     public List<Comment> getAll() {
-        return null;
+        Query<Comment> query;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java:49]: why are you declaring this variable outside of `try`?

File: src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java [hunk 8]
```diff
@@ -12,21 +15,64 @@ public CommentDaoImpl(SessionFactory sessionFactory) {
...
+        try (Session session = factory.openSession()) {
+            query = session.createQuery("SELECT e FROM Comment e", Comment.class);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java:51]:
```suggestion
            query = session.createQuery("FROM Comment", Comment.class);
```
let's simplify all `getAll()` queries like this

File: src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java [hunk 9]
```diff
@@ -12,21 +15,64 @@ public CommentDaoImpl(SessionFactory sessionFactory) {
...
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get all comment", e);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java:54]:
```suggestion
            throw new RuntimeException("Can't get all comments", e);
```
same everywhere

File: src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java [hunk 10]
```diff
@@ -12,16 +15,42 @@ public SmileDaoImpl(SessionFactory sessionFactory) {
...
+        }
     }
 
     @Override
     public Smile get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.get(Smile.class, id);
+        } catch (Exception e) {
+            throw new RuntimeException("Smile with id= " + id + "doesn't exits", e);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java:43]: in all your classes you have different exception messages for same operations

it looks not consistent - let's choose one style and apply it to all messages

File: src/main/java/core/basesyntax/dao/impl/UserDaoImpl.java [hunk 11]
```diff
@@ -12,21 +15,63 @@ public UserDaoImpl(SessionFactory sessionFactory) {
...
 
     @Override
     public void remove(User entity) {
-
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.remove(entity);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/dao/impl/UserDaoImpl.java:64]: if you decided so - let's use Hibernate methods everywhere

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**